### PR TITLE
Wrap variable name in implicit_dictionary with `to_lower()`

### LINKED ISSUE #1378: SciPy's specfun.f
There is only one issue:
```console
$ lfortran --show-asr --fixed-form --implicit-typing --implicit-interface ./scipy/special/specfun/specfun.f
semantic error: Arguments do not match for any generic procedure, dimag
     --> ./scipy/special/specfun/specfun.f:11559:13
      |
11559 |         IF (DIMAG(Z).EQ.0.0D0.OR.XC.LT.1.0D0) LS=1
      |             ^^^^^^^^ 
```
Otherwise it compiles to ASR. LLVM then has this issue:
```console
$ lfortran --show-llvm --fixed-form --implicit-typing --implicit-interface ./scipy/special/specfun/specfun.f
code generation error: ComplexConstructor with runtime arguments not implemented yet.
```

### PR DESCRIPTION
**Pull request by Pranavchiku**
Fixes #1378.

### CODE REVIEW

**Comment by Pranavchiku** [comment]
I am not sure whether this is correct fix or not, it passes the test registered, but it still gives same error for `specfun.f`.

**Comment by certik** [comment]
> it still gives same error for specfun.f.

What error do you get?

**Comment by certik** [comment]
The ASR tests changed for other tests with this PR, so we need to carefully investigate. Was there a bug before, or did we introduce a bug with this PR?

**Comment by Pranavchiku** [comment]
There was a bug before, this PR fixed the registered test, and hence must generate ASR for `specfun.f` but it is not generating, I will have a close look at it again. Pardon me for delaying to respond.

**Comment by Pranavchiku** [comment]
With the latest commit, it now fixes #1378, and compiles `scipy/specfun.f` to ASR.

File: tests/reference/asr-implicit6-d68cc03.stdout [hunk 1]
```diff
@@ -71,7 +71,7 @@
                     ch12n
                     (FunctionType
                         [(Integer 4 [])
-                        (Real 4 [])]
+                        (Complex 8 [])]
```
Comment (Pranavchiku) [inline tests/reference/asr-implicit6-d68cc03.stdout:74]: These changes looks good to me, previously they were typed wrongly.

File: tests/reference/asr-implicit5-0330703.stdout [hunk 2]
```diff
@@ -126,7 +126,7 @@
                                     ()
                                     ()
                                     Default
-                                    (Real 4 [])
+                                    (Real 8 [])
```
Comment (Pranavchiku) [inline tests/reference/asr-implicit5-0330703.stdout:129]: These also look good to me.

**Review by czgdp1807 — APPROVED** [review]
We should figure out a way to convert to lower case for any string so that these kind of fixes aren't needed in the future.

**Comment by certik** [comment]
There might be a bug with `is_Subroutine` if the visit_Function becomes nested (such as visiting the "contains" section), but we can rework it later if that is the case.

**Review by certik — APPROVED** [review]

**Comment by certik** [comment]
The changes in the reference tests are good, they were wrong before, now they are right.

**Comment by certik** [comment]
> We should figure out a way to convert to lower case for any string so that these kind of fixes aren't needed in the future.

Yes, we should. One way is to convert everything to lowercase in the prescanner, but I was hoping to eventually be able to give warnings/errors for inconsistent case for variables (declaring as `A`, using as `a`). But doing some systematic way would be great.

**Pull request merged**